openhtf/io/output/json_factory.py

```diff
@@ -30,33 +26,19 @@ class OutputToJSON(JSONEncoder):
   """
 
   def __init__(self, filename_pattern=None, inline_attachments=True, **kwargs):
-    super(OutputToJSON, self).__init__(**kwargs)
-    self.filename_pattern = filename_pattern
+    super(OutputToJSON, self).__init__(filename_pattern)
     self.inline_attachments = inline_attachments
-
-  def default(self, obj):
-    if isinstance(obj, BaseException):
-      # Just repr exceptions.
-      return repr(obj)
-    return super(OutputToJSON, self).default(obj)
-
-  # pylint: disable=invalid-name
-  def __call__(self, test_record):
-    assert self.filename_pattern, 'filename_pattern required'
+    self.json_encoder = JSONEncoder(**kwargs)
+ 
+  def serialize_test_record(self, test_record):
+    return self.json_encoder.encode(self.ConvertToDict(test_record))
+  
+  def ConvertToDict(self, test_record):
     if self.inline_attachments:
       as_dict = data.ConvertToBaseTypes(test_record)
       for phase in as_dict['phases']:
         for value in phase['attachments'].itervalues():
           value['data'] = base64.standard_b64encode(value['data'])
     else:
       as_dict = data.ConvertToBaseTypes(test_record, ignore_keys='attachments')
```

Comment:
here is where it needs to be `('attachments',)` instead... looks like it was a pre-existing bug that we just never caught.